oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/manual/process/ProcessTraffic.java

```diff
@@ -0,0 +1,226 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.analysis.manual.process;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.skywalking.oap.server.core.Const;
+import org.apache.skywalking.oap.server.core.analysis.IDManager;
+import org.apache.skywalking.oap.server.core.analysis.Layer;
+import org.apache.skywalking.oap.server.core.analysis.MetricsExtension;
+import org.apache.skywalking.oap.server.core.analysis.Stream;
+import org.apache.skywalking.oap.server.core.analysis.metrics.Metrics;
+import org.apache.skywalking.oap.server.core.analysis.worker.MetricsStreamProcessor;
+import org.apache.skywalking.oap.server.core.remote.grpc.proto.RemoteData;
+import org.apache.skywalking.oap.server.core.storage.StorageHashMapBuilder;
+import org.apache.skywalking.oap.server.core.storage.annotation.Column;
+import org.apache.skywalking.oap.server.library.util.StringUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.skywalking.oap.server.core.source.DefaultScopeDefine.PROCESS;
+
+@Stream(name = ProcessTraffic.INDEX_NAME, scopeId = PROCESS,
+    builder = ProcessTraffic.Builder.class, processor = MetricsStreamProcessor.class)
+@MetricsExtension(supportDownSampling = false, supportUpdate = true)
+@EqualsAndHashCode(of = {
+        "instanceId",
+        "name",
+})
+public class ProcessTraffic extends Metrics {
+    public static final String INDEX_NAME = "process_traffic";
+    public static final String SERVICE_ID = "service_id";
+    public static final String INSTANCE_ID = "instance_id";
+    public static final String NAME = "name";
+    public static final String LAYER = "layer";
+    public static final String AGENT_ID = "agent_id";
+    public static final String PROPERTIES = "properties";
+    public static final String LAST_PING_TIME_BUCKET = "last_ping";
+    public static final String DETECT_TYPE = "detect_type";
+
+    private static final Gson GSON = new Gson();
+
+    @Setter
+    @Getter
+    @Column(columnName = SERVICE_ID)
+    private String serviceId;
+
+    @Setter
+    @Getter
+    @Column(columnName = INSTANCE_ID)
+    private String instanceId;
+
+    @Setter
+    @Getter
+    @Column(columnName = NAME)
+    private String name;
+
+    @Setter
+    @Getter
+    @Column(columnName = LAYER)
+    private int layer = Layer.UNDEFINED.value();
+
+    @Setter
+    @Getter
+    @Column(columnName = LAST_PING_TIME_BUCKET)
+    private long lastPingTimestamp;
+
+    @Setter
+    @Getter
+    @Column(columnName = DETECT_TYPE)
+    private int detectType = ProcessDetectType.UNDEFINED.value();
+
+    @Setter
+    @Getter
+    @Column(columnName = AGENT_ID)
```

Comment:
The default length is only 200, is this enough?

Reply:
update to 500 length.